recipes/newrelic/infrastructure/amazonlinux.yml

```diff
@@ -100,18 +100,16 @@ install:
             sudo sed -i "/^enable_process_metrics/d" /etc/newrelic-infra.yml
             sudo sed -i "/^status_server_enabled/d" /etc/newrelic-infra.yml
             sudo sed -i "/^status_server_port/d" /etc/newrelic-infra.yml
+            sudo sed -i "/^license_key/d" /etc/newrelic-infra.yml
 
-            sudo sed -i "s/^\(license_key: \).*/\1{{.NEW_RELIC_LICENSE_KEY}}/" /etc/newrelic-infra.yml
           else
             sudo touch /etc/newrelic-infra.yml
-            sudo cat > /etc/newrelic-infra.yml <<EOT
-          license_key: {{.NEW_RELIC_LICENSE_KEY}}
-          EOT
           fi
         - |
           if [ $(echo {{.NEW_RELIC_REGION}} | grep -i staging | wc -l) -gt 0 ]; then
             sudo echo 'staging: true' >> /etc/newrelic-infra.yml
           fi
+          sudo sed -i "1s/^/license_key: {{.NEW_RELIC_LICENSE_KEY}}\n/" /etc/newrelic-infra.yml
```

Comment:
why do you use sed here instead of echoing the line?

Reply:
I just wanted to keep the same structure with license on top of config, it can be echo to the end as well

Reply:
Could we keep it consistent with the other echo below, it's a bit puzzling for a reader to understand why licenseKey is handled differently.